services/slam/fake/slam_test.go

```diff
@@ -0,0 +1,177 @@
+	test.That(t, pInFrame.Pose().Point(), test.ShouldResemble, expectedPoint)
+
+	expectedOri := spatialmath.NewR4AA()
+	expectedOri.RZ = -1
+	expectedOri.Theta = 3.0542483867902197
+	test.That(t, pInFrame.Pose().Orientation().AxisAngles(), test.ShouldResemble, expectedOri)
+
+	pInFrame2, err := slamSvc.Position(context.Background(), slamSvc.Name, map[string]interface{}{})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pInFrame2, test.ShouldResemble, pInFrame2)
+}
+
+func TestFakeSLAMGetInternalState(t *testing.T) {
+	slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+	data, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, len(data), test.ShouldBeGreaterThan, 0)
+	data2, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, data, test.ShouldResemble, data2)
+}
+
+func TestFakeSLAMStateful(t *testing.T) {
+	t.Run("Test getting a PCD map advances the test data", func(t *testing.T) {
+		slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+		extra := map[string]interface{}{}
+		verifyGetMapStateful(t, rdkutils.MimeTypePCD, slamSvc, extra)
+	})
+}
+
+func TestFakeSLAMGetMap(t *testing.T) {
+	extra := map[string]interface{}{}
+
+	t.Run("Test getting valid JPEG map", func(t *testing.T) {
+		slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+		pInFrame := referenceframe.NewPoseInFrame(
+			slamSvc.Name,
+			spatialmath.NewPose(
+				r3.Vector{X: 0, Y: 0, Z: 0},
+				spatialmath.NewOrientationVector(),
+			),
+		)
+		mimeType, im, vObj, err := slamSvc.GetMap(
+			context.Background(),
+			slamSvc.Name,
+			rdkutils.MimeTypeJPEG,
+			pInFrame,
+			true,
+			extra,
+		)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, mimeType, test.ShouldEqual, rdkutils.MimeTypeJPEG)
+		test.That(t, vObj, test.ShouldBeNil)
+		test.That(t, im, test.ShouldNotBeNil)
+		test.That(t, im.Bounds().Max.X, test.ShouldEqual, 1909)
+		test.That(t, im.Bounds().Max.Y, test.ShouldEqual, 4876)
+		test.That(t, im.Bounds().Min.X, test.ShouldEqual, 0)
+		test.That(t, im.Bounds().Min.Y, test.ShouldEqual, 0)
+	})
+
+	t.Run("Test getting invalid PNG map", func(t *testing.T) {
+		slamSvc := &SLAM{Name: "test", logger: golog.NewTestLogger(t)}
+		pInFrame := referenceframe.NewPoseInFrame(
+			slamSvc.Name,
+			spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0},
+				spatialmath.NewOrientationVector()),
+		)
+		mimeType, im, vObj, err := slamSvc.GetMap(
+			context.Background(),
+			slamSvc.Name,
+			rdkutils.MimeTypePNG,
+			pInFrame,
+			true,
+			extra,
+		)
+		test.That(t, err, test.ShouldBeError, "received invalid mimeType for GetMap call")
+		test.That(t, mimeType, test.ShouldEqual, "")
+		test.That(t, vObj, test.ShouldBeNil)
+		test.That(t, im, test.ShouldBeNil)
+	})
+}
+
+func verifyGetMapStateful(t *testing.T, mimeType string, slamSvc *SLAM, extra map[string]interface{}) {
```

Comment:
I removed the image tests from this as they were timing out in CI & we are not going to need them after the Map Representation project.